test/e2e/test_job_user.py

```diff
@@ -100,18 +73,13 @@ def test_job_run_as_user(
         # THEN
         job.wait_until_complete(client=deadline_client, max_retries=20)
 
-        # Retrieve job output and verify whoami printed the queue's jobsRunAsUser
-        job_logs = job.get_logs(
+        job.assert_single_task_log_contains(
             deadline_client=deadline_client,
             logs_client=boto3.client(
                 "logs",
                 config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
             ),
+            expected_pattern=r"{job_run_as_user.user}",
```

Comment:
I think we want:

```suggestion
            expected_pattern=r"I am: {re.escape(job_run_as_user.user)}",
```

Rationale:

1.  The worker agent logs which user it *INTENDS* to run the session actions as in the session log, and this will match that logging output. We want to be sure that the job is actually running as the user so we want to match the "I am: " part.
2. The [`re.escape(...)`](https://docs.python.org/3/library/re.html#re.escape) function ensures the username is matched verbatim and no regex escape characters will be interpreted incorrectly in the pattern.

Reply:
Updated